mongo/database.go

```diff
@@ -58,13 +58,18 @@ func newDatabase(client *Client, name string, opts ...*options.DatabaseOptions)
 		wc = dbOpt.WriteConcern
 	}
 
+	rg := client.registry
+	if dbOpt.Registry != nil {
+		rg = dbOpt.Registry
+	}
+
 	db := &Database{
 		client:         client,
 		name:           name,
 		readPreference: rp,
 		readConcern:    rc,
 		writeConcern:   wc,
-		registry:       client.registry,
+		registry:       rg,
```

Comment:
If you revert this change does the new test still pass?

Reply:
Yes, a method has been added to compare registries in the test file that would fail without the newest change.

Reply:
Yes, looks better now!